org.eclipse.languageserver/src/org/eclipse/languageserver/operations/completion/LSContentAssistProcessor.java

```diff
@@ -41,10 +41,10 @@ public LSContentAssistProcessor() {
 		final LSPDocumentInfo info = LanguageServiceAccessor.getLSPDocumentInfoFor(viewer, ServerCapabilities::isCodeActionProvider);
 		CompletableFuture<CompletionList> request = null;
 		try {
-			if (info.languageClient != null) {
+			if (info.hasLanguageClient()) {
```

Comment:
Should an info without a language client even be allowed? If we add the good assertions and Nullable annotation to LSPDocumentInfo, then we shouldn't have to check for that.